src/client/entity-editor/name-section/name-section.js

```diff
@@ -149,12 +157,24 @@ class NameSection extends React.Component {
 
 		const warnIfExists = !_.isEmpty(exactMatches);
 
+		if (!this.state.showButton && searchResults && searchResults.length > 5) {
+			this.setState({showButton: true});
+		}
+
+		if (this.state.showButton && searchResults && searchResults.length <= 5) {
+			this.setState({showButton: false});
+		}
+
+		if (this.state.showButton && searchResults === null) {
+			this.setState({showButton: false});
+		}
+
```

Comment:
This can be refactored into:
```suggestion
		if (this.state.showButton && searchResults?.length <= 5) {
			this.setState({showButton: false});
		}
		else {
			this.setState({showButton: true});
		}

```

In the expression `searchResults?.length <= 5`, the `?.` is a fairly new javascript feature called optional chaining.
If `searchResult` is null or undefined, the whole expression will evaluate as false